server/cluster.go

```diff
@@ -1293,11 +1258,32 @@ func (c *Cluster) garbageCollectProxySessions(activeNodes []string) {
 	}
 }
 
+// cleanUpRemoteSubs adjusts online user counts in the master (proxied) topic
+// for a leaving proxy session.
+func (sess *Session) cleanUpRemoteSessions(topicName string) {
+	sess.remoteSessionsLock.Lock()
+	defer sess.remoteSessionsLock.Unlock()
```

Comment:
This enables a deadlock on `remoteSessionsLock` and `hub.topicGet`. Instead of `defer .. Unlock()` I would unlock before `if t := globals.hub.topicGet(topicName); t != nil {`

Reply:
True. Thanks.